syft/frameworks/torch/tensors/interpreters/native.py

```diff
@@ -1042,16 +1042,44 @@ def encrypt(self, protocol="mpc", **kwargs):
                 "Encryption and Secure Multi-Party Computation"
             )
 
-    def decrypt(self, private_key):
-        """This method will decrypt each value in the tensor, returning a normal
-        torch tensor.
+    def decrypt(self, protocol="mpc", **kwargs):
+        """
+        This method will decrypt each value in the tensor using Multi Party
+        Computation (default) or Paillier Homomorphic Encryption
 
         Args:
-            *private_key a private key created using
-                syft.frameworks.torch.he.paillier.keygen()
-            """
+            protocol (str): Currently supports 'mpc' for Multi Party
+                Computation and 'paillier' for Paillier Homomorphic Encryption
+            **kwargs:
+                With Respect to MPC accepts:
+                    None
+
+                With Respect to Paillier accepts:
+                    private_key (phe.paillier.PaillierPrivateKey): Can be obtained using
+                        ```public_key, private_key = sy.frameworks.torch.he.paillier.keygen()```
+        Returns:
+            An decrypted version of the Tensor following the protocol specified
+
+        Raises:
+            NotImplementedError: If protocols other than the ones mentioned above are queried
+
+        """
+
+        if protocol.lower() == "mpc":
+            x_encrypted = self.copy()
```

Comment:
Do you need  copy() here?

Reply:
It served a purpose of not making it an inplace operation

Reply:
Oh I see 
Ok I got your point, let's keep it your way